core/src/OpenMP/Kokkos_OpenMP_Instance.hpp

```diff
@@ -139,28 +147,24 @@ inline bool OpenMP::impl_is_initialized() noexcept {
   return Impl::OpenMPInternal::singleton().is_initialized();
 }
 
-inline bool OpenMP::in_parallel(OpenMP const&) noexcept {
-  // FIXME_OPENMP We are forced to use t_openmp_instance because the function is
-  // static and does not use the OpenMP object
-  return ((Impl::OpenMPInternal::singleton().m_level < omp_get_level()) &&
-          (!Impl::t_openmp_instance ||
-           Impl::t_openmp_instance->m_level < omp_get_level()));
+inline bool OpenMP::in_parallel(OpenMP const& exec_space) noexcept {
+  return (
+      (exec_space.impl_internal_space_instance()->m_level < omp_get_level()) &&
+      (!Impl::t_openmp_instance ||
+       Impl::t_openmp_instance->m_level < omp_get_level()));
 }
 
-inline int OpenMP::impl_thread_pool_size() noexcept {
-  // FIXME_OPENMP We are forced to use t_openmp_instance because the function is
-  // static
-  return OpenMP::in_parallel()
+inline int OpenMP::impl_thread_pool_size(OpenMP const& exec_space) noexcept {
+  return OpenMP::in_parallel(exec_space)
              ? omp_get_num_threads()
              : (Impl::t_openmp_instance
```

Comment:
In which cases do we actually need to use `t_openmp_instance`? Is that only necessary for `partition_master`?

Reply:
Yes, it's only for `partition_master`